### DISCUSSION TITLE
Bug: (Implementing no-restricted-exports rule as described in docs throws error)

### DISCUSSION THREAD

**Discussion by Distance71** [eslint/eslint#15487] [Q&A]
### Environment

Node version: v16.13.0
npm version: v8.1.0
Local ESLint version: v6.8.0 and la v8.6.0
Global ESLint version: Not found
Operating System: MacOs Big Sur

### What parser are you using?

@typescript-eslint/parser

### What did you do?

```js
module.exports = {
  env: {
    browser: true,
    mocha: true,
    jest: true
  },
  parser: "@typescript-eslint/parser", // Specifies the ESLint parser
  parserOptions: {
    project: "./tsconfig.json",
    tsconfigRootDir: ".",
    ecmaVersion: 2019, // Allows for the parsing of modern ECMAScript features
    sourceType: "module", // Allows for the use of imports
    ecmaFeatures: {
      jsx: true // Allows for the parsing of JSX
    }
  },
  rules: {
    "no-restricted-exports": ["error", { restrictedNamedExports: ["default"] }],
  },
  settings: {
    react: {
      version: "detect" // Tells eslint-plugin-react to automatically detect the version of React to use
    },
    "import/extensions": [".js", ".jsx", ".ts", ".tsx"],
    "import/parsers": {
      "@typescript-eslint/parser": [".ts", ".tsx"]
    },
    "import/resolver": {
      node: {
        extensions: [".js", ".jsx", ".ts", ".tsx"]
      }
    }
  }
};
```

### What did you expect to happen?

Rule works as expected [in docs](https://eslint.org/docs/rules/no-restricted-exports) (`"no-restricted-exports": ["error", { restrictedNamedExports: ["default"] }]`)

### What actually happened?

`error  Definition for rule 'no-restricted-exports' was not found  no-restricted-exports`

**Comment by ljharb** [comment]
How are you running eslint?

**Comment by Distance71** [comment]
`yarn eslint . --ext .ts,.tsx --quiet`

**Comment by ljharb** [comment]
and what's `yarn eslint --version`?

**Comment by Distance71** [comment]
I tried with the two described in the environment section, v6.8.0 and v8.6.0

**Comment by aladdin-add** [comment]
This rule was introduced in ESLint 7.0.0-alpha.0. You need to upgrade your eslint to use it.
